frontend/src/components/Calendar/Day.js

```diff
@@ -1,28 +1,41 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import styles from "./Day.module.css";
 import dayjs from "dayjs";
 import { useSelector, useDispatch } from 'react-redux'; //To manage Global State of Redux
 import { toggle } from '../../slices/ShowModalSlice' //Import toggle function to turn on/off Modal
 
+
 function Day({ day, row }) {
-  const ShowModal = useSelector(state => state.ShowModal.value); //ShowModal is a boolean state that know as True - showing and False - not showing
+  const currentDate = new Date(day)
   const dispatch = useDispatch(); //dispatch is to use function to interact with State of Redux
   const isToday = day.isSame(dayjs(), 'day');  // Check if the day is today
   const circleColor = isToday ? "#aacaef" : ""; // If it is today, the circle will be blue
 
-  const handleClick = () => {
-    alert(day.format("DD-MM-YY"));
-  };
+  const MonthEvents = useSelector(state => state.MonthEvents.value)
+  const todayEvents = MonthEvents.filter(event => {  //Algorithm to filter which events is on today
+    const start = new Date(event.start);
+    const end = new Date(event.end);
+    const startOfCurrentDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 0, 0, 0, 1);
+    const endOfCurrentDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 23, 59, 59, 999);
+    const minTime = Math.min(start.getTime(), end.getTime(), startOfCurrentDate.getTime(), endOfCurrentDate.getTime())
+    const maxTime = Math.max(start.getTime(), end.getTime(), startOfCurrentDate.getTime(), endOfCurrentDate.getTime())
+    return ((maxTime - minTime) < 24 * 60 * 60 * 1000 + (end.getTime() - start.getTime()));
+  });

```

Comment:
It's a good practice to separate the logic from the rendering, so best to extract it to a helper/util file

Reply:
Got it, I've done with them